src/CodeGen_C.cpp

```diff
@@ -1235,9 +1250,19 @@ void CodeGen_C::visit(const Call *op) {
             rhs << "_ucon, ";
         }
 
+        // TODO: special-case for https://github.com/halide/Halide/issues/2099;
+        // embedded buffers will generate code for bounds query, which can
+        // never execute (_halide_buffer_is_bounds_query() always returns false
+        // for them), but the generated bounds-query code won't compile due to 
+        // const correctness. We add some const_casts here as a temporary
+        // workaround.
```

Comment:
Well... we know they should never be mutated, so it's a good idea to put them in the .rodata section and fail early if someone tries to mess with one. I don't think we can get that nuance into the C type system though. This thing has to be of a mutable type (because we pass it to functions that *might* mutate that argument in the general case), but this specific object shouldn't be mutated by those functions based on some properties I happen to know about it, so I want to mprotect it.

I guess the const_cast adequately captures this, but it wouldn't be a temporary workaround.